include/cutlass/gemm/kernel/intel_pvc_gemm.hpp

```diff
@@ -215,22 +212,28 @@ class GemmUniversal<
 
     // Get the appropriate blocks for this sub_group -- potential for sub_group locality
     int thread_idx = int(ThreadIdxX());
+    int sub_group_id = thread_idx / SubgroupSize;
     constexpr auto workgroup_shape = WorkgroupTileShape{};                                                  // (SUB_M,SUB_N,SUB_K)
     constexpr auto subgroup_shape = SubgroupTileShape{};                                                  // (SUB_M,SUB_N,SUB_K)
-    const int m_coord = BlockIdxX() * get<0>(subgroup_shape);
-    const int n_coord = BlockIdxY() * get<1>(workgroup_shape) + thread_idx / SubgroupSize * get<1>(subgroup_shape);
+    const int m_coord = BlockIdxX() * get<0>(workgroup_shape) + sub_group_id / CollectiveMainloop::sg_per_wg_n * get<0>(subgroup_shape);
+    const int n_coord = BlockIdxY() * get<1>(workgroup_shape) + sub_group_id % CollectiveMainloop::sg_per_wg_n * get<1>(subgroup_shape);
     const int l_coord = BlockIdxZ();
     const auto tile_coord = make_coord(m_coord, n_coord, _, l_coord);
 
     Tensor tAi = params.mainloop.gmem_tiled_copy_a.get_pvc_tensor(
             make_coord(m_coord, 0, 0),
             make_shape(_1{}, K, L),
             make_stride(Int<FragsM>{} * get<0>(MmaAtomShape()),_1{}));
+    constexpr int version =
+        is_same_v<typename CollectiveMainloop::GmemTiledCopyB,
+                  XE_2D_U16x16x16x2x1_V>
+            ? 1
+            : 2;
 
     Tensor tBi = params.mainloop.gmem_tiled_copy_b.get_pvc_tensor(
-            make_coord(n_coord, 0, 0),
-            make_shape(Int<FragsN>{}, K / 2, L),
-            make_stride(get<1>(MmaAtomShape()), _1{}));
+            make_coord(0, n_coord, 0),
+            make_shape(K, Int<FragsN / version>{}, L),
+            make_stride(_1{}, Int<version * get<1>(MmaAtomShape())>{}));
```

Comment:
I think this is reverting the changes to define this tensor in N-major order, and it is now back to row-major order. We must use N-major as it is required by the design in Cutlass. Row-major and column major will be inferred from the stride.